client/components/fields/editor/base/text.tsx

```diff
@@ -58,8 +81,16 @@ export class EditorFieldText extends React.Component<IEditorFieldTextProps, ISta
         }
     }
 
+    componentDidUpdate = (prevProps: Readonly<IEditorFieldTextProps>): void => {
+        // Make sure to reset user modification state when item changes
+        // so that late-arriving prop values can populate the field again
+        if (!isEqual(prevProps.item, this.props.item)) {
+            this.setState({userHasModified: false});
```

Comment:
I wonder now why not set it in the `getDerivedStateFromProps` ?

Reply:
but thinking it twice, I think I could solve it with `getDerivedStateFromProps`. Let me give it a try.

Reply:
@petrjasek nope, my idea did not work. It prevents the user from typing.